jobdsl-gradle/src/main/groovy/job/JobBuilder.groovy

```diff
@@ -172,7 +172,24 @@ public class JobBuilder {
         }
         this
     }
-
+    /**
+     * Use this function to add Build trigger
+     *
+     * @ArrayList<String>    projects List of upsteam projects trigger will watch
+     * @String               threshold the result of the upstreams build when run the job.
+     *                       SUCCESS by default
+     */
+        public JobBuilder addBuildTrigger(ArrayList<String> projects, String threshold = 'SUCCESS', boolean debug = false) {
```

Comment:
This is a too generic name - you are adding a very specific type of trigger. BTW, do we really need to wrap such one-liners? Would it be possible to just update the object when it is created